src/components/AddTask.jsx

```diff
@@ -1,24 +1,40 @@
 import { TextField } from "@mui/material";
-import { useCallback, useState } from "react";
+import { memo, useCallback, useState } from "react";
+import useStore from "./../store/customHook";
 
-function AddTask({ onAddTodoList }) {
+const AddTask = () => {
+  const [todoList, dispatch] = useStore();
   const [title, setTitle] = useState("");
+  const [taskId, setTaskId] = useState(0);
 
   const handleSubmit = useCallback(
     (e) => {
       e.preventDefault();
-      onAddTodoList(title);
+      if (title) {
+        dispatch({
+          type: "add",
+          payload: {
+            taskId,
+            name: title,
+            status: "pendding",
+          },
+        });
+        setTaskId(taskId + 1);
```

Comment:
Why not use UUID instead of increasing the taskId here? If we initiate the list of tasks from the database. It would be a problem because the initial value of the state is 0.